src/reset.c

```diff
@@ -48,11 +48,31 @@ void reset_reset(bool status)
 {
     keystore_lock();
 #if !defined(TESTING)
-    if (!securechip_update_keys()) {
+    int retries_update_keys = 0;
+    const int max_retries_update_keys = 5;
+    bool sc_result_update_keys = false;
+    while (retries_update_keys < max_retries_update_keys) {
+        sc_result_update_keys = securechip_update_keys();
+        if (sc_result_update_keys) {
+            break;
+        }
+        retries_update_keys++;
+    }
+    if (!sc_result_update_keys) {
         Abort("Could not reset secure chip.");
```

Comment:
sorry for one more nit but i now realized that all these loops are simply for loops:

```
bool sc_result_update_keys = false;
    for (int retries = 0; retries < 5; retries++) {
        sc_result_update_keys = securechip_update_keys();
        if (sc_result_update_keys) {
            break;
        }
    }
    if (!sc_result_update_keys) {
        Abort("Could not reset secure chip.");
    }
```

(in this case no need for a different retry var name as it is locally scoped to the for loop).

same in securechip.c